File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JsonFieldObjectWrapper.java
package org.codinjutsu.tools.nosql.commons.view.add;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import org.codinjutsu.tools.nosql.commons.model.internal.layer.DatabaseElement;

import static org.codinjutsu.tools.nosql.commons.model.internal.json.JsonHelperKt.convert;

public class JsonFieldObjectWrapper extends JTextFieldWrapper {

    @Override
    public DatabaseElement getValue() {
        return convert(new Gson().fromJson(component.getText(), JsonObject.class));
    }
}

File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JsonFieldObjectWrapper.kt
package org.codinjutsu.tools.nosql.commons.view.add

import com.google.gson.Gson
import com.google.gson.JsonObject
import com.intellij.openapi.project.Project
import org.codinjutsu.tools.nosql.commons.model.internal.toDatabaseElement

internal class JsonFieldObjectWrapper(project: Project) : JsonEditorWrapper(project) {

    override fun getValue() =
            Gson().fromJson(editor.document.text, JsonObject::class.java).toDatabaseElement()

    override fun reset() {
        editor.document.setText("{}")
    }
}
